src/assets/scripts/client/InputController.js

```diff
@@ -722,51 +563,98 @@ export default class InputController {
+                this.$commandInput.attr('placeholder', 'enter scope command');
+                // this.$commandInput.css({'background-color': '#444'});
+                this.$commandInput.css({ color: 'red' });
+
+                return;
+
+            case COMMAND_CONTEXT.SCOPE:
+                this.commandBarContext = COMMAND_CONTEXT.AIRCRAFT;
+                this.$commandInput.attr('placeholder', 'enter aircraft command');
+                // this.$commandInput.css({ 'background-color': '#999' });
+                this.$commandInput.css({ color: 'white' });
+
+                return;
+
+            default:
+                return;
+        }
+    }
+
+    /**
+     * Process user command to be applied to an aircraft
+     *
+     * @for InputController
+     * @method processAircraftCommand
+     */
+    processAircraftCommand() {
+        //     this.input.history.unshift(this.input.callsign);
+        //     this.$commandInput.val('');
+        //     this.input.command = '';
+
+        let aircraftCommandParser;
         // this could use $commandInput.val() as an alternative
         const userCommand = this.input.command.trim().toLowerCase();
 
-        // Using try/catch here very much on purpose. the `CommandParser` will throw when it encounters any kind
+        // Using try/catch here very much on purpose. the `AircraftCommandParser` will throw when it encounters any kind
         // of error; invalid length, validation, parse, etc. Here we catch those errors, log them to the screen
         // and then throw them all at once
         try {
-            result = new CommandParser(userCommand);
+            aircraftCommandParser = new AircraftCommandParser(userCommand);
         } catch (error) {
             UiController.ui_log('Command not understood');
 
             throw error;
         }
 
-        return result;
+        if (aircraftCommandParser.command !== 'transmit') {
+            return this.processSystemCommand(aircraftCommandParser);
+        }
+
+        return this.processTransmitCommand(aircraftCommandParser);
     }
 
     /**
+     * Process user command to be applied to the user's scope
+     *
      * @for InputController
-     * @method input_run
+     * @method processScopeCommand
      */
-    input_run() {
-        const commandParser = this._parseUserCommand();
+    processScopeCommand() {
+        let scopeCommandModel;
+        const userCommand = this.input.command.trim().toLowerCase();
 
-        if (commandParser.command !== 'transmit') {
-            return this.processSystemCommand(commandParser);
+        try {
+            scopeCommandModel = new ScopeCommandModel(userCommand);
+        } catch (error) {
+            UiController.ui_log('ERROR: BAD SYNTAX');
+
+            throw error;
         }
 
-        return this.processTransmitCommand(commandParser);
+        const [successful, response] = this._scopeModel.runScopeCommand(scopeCommandModel);
```

Comment:
I'm not sure what `successful` is here?  especially since the next statement sends `!successful` as an argument.  What is this actually doing?

Reply:
Whether the command was understood and executed correctly. The opposite issue is because the `UiController.ui_log()` function takes arguments `message` and `isWarning`.

I did:
```
const isWarning = !successful;

UiController.ui_log(response, isWarning);
```

That should provide more clarity.